transformer_engine/jax/layernorm.py

```diff
@@ -162,6 +164,12 @@ def _layernorm_fp8_dot_fwd_rule(
     k_contracting_dims = (0,)
     assert x.shape[-1] == kernel.shape[0]
 
+    is_fp8_meta_fm32 = amax.dtype == FP8MetaFP32
+    if is_fp8_meta_fm32:
+        fp8_max, amax, scale, scale_inv = \
+            convert_fp8_meta_fm32tofp32(fp8_max, amax, scale, scale_inv)
+
+    scale, scale_inv = FP8Helper.update_fp8_scale(fp8_max, amax, scale)
```

Comment:
Explicitly calling the conversion here is because the conversion is for whole functions, instead of `update_fp8_scale` only. Explicitly calling the conversion ahead can make sure the dtype of FP8 meta inside the function (like `layernorm_fp8_dot`) would be the FP32, regardless the order to the computation and `update_fp8_scale`, like
```python
conversion
computation # Require FP32
update_scale # Require FP32
```

Reply:
That's a valid point. You're right, we need these separated.